lib/foreman/util.rb

```diff
@@ -22,9 +22,10 @@ def secure_token
       SecureRandom.base64(96).tr('+/=', '-_*')
     end
 
-    # recommended to make encryption_key at least 32 bytes. Ex. SecureRandom.hex(20)
+    # recommended to make encryption_key 32 bytes, matching the key length preferred by
+    # AS::MessageEncryptor's default algorithm
     def secure_encryption_key
-      SecureRandom.hex(20)
+      SecureRandom.hex(ActiveSupport::MessageEncryptor.key_len / 2)
```

Comment:
Sure, just for the record, hexadecimal key entropy is substantially smaller (4.000) than upper&lower case alphabet (5.7004). Since base64 is even more (three more characters) than just alphabet, it's approximetly half-times stronger (4 vs 6).

The failure is irrelevant, thanks.